extensions-core/s3-extensions/src/main/java/io/druid/storage/s3/S3DataSegmentPuller.java

```diff
@@ -135,7 +144,8 @@ public void finalize() throws Throwable
       {
         try {
           if (!streamAcquired) {
-            s3Obj.closeDataInputStream();
+            // Make absolutely sure
```

Comment:
If the stream wasn't acquired then we don't have to close it on finalization anymore, due to lazy opening. So I think we never need to do this.